src/components/CompanySwitcher.tsx

```diff
@@ -13,25 +13,53 @@ import { AuthenticationTypes } from 'common/dtos/authentication';
 import { useCurrentUser } from 'common/hooks/useCurrentUser';
 import { authenticate } from 'common/stores/slices/user';
 import { RootState } from 'common/stores/store';
-import { Fragment } from 'react';
-import { Check, ChevronDown } from 'react-feather';
+import { Fragment, useEffect, useState } from 'react';
+import { Check, ChevronDown, PlusCircle } from 'react-feather';
 import { useTranslation } from 'react-i18next';
 import { useQueryClient } from 'react-query';
 import { useDispatch, useSelector } from 'react-redux';
 import { route } from 'common/helpers/route';
 import { DropdownElement } from './dropdown/DropdownElement';
 import { useLogo } from 'common/hooks/useLogo';
 import { useCompanyName } from 'common/hooks/useLogo';
+import { CompanyCreate } from 'pages/settings/company/create/CompanyCreate';
+import { CompanyEdit } from 'pages/settings/company/edit/CompanyEdit';
+import { useCurrentCompany } from 'common/hooks/useCurrentCompany';
+import { isHosted, isSelfHosted } from 'common/helpers';
+import { freePlan } from 'common/guards/guards/free-plan';
 
 export function CompanySwitcher() {
   const [t] = useTranslation();
+
   const user = useCurrentUser();
+
   const state = useSelector((state: RootState) => state.companyUsers);
+
+  const canUserAddCompany = isSelfHosted() || (isHosted() && !freePlan());
+
+  const isCompanyEditAlreadyOpened = sessionStorage.getItem(
+    'COMPANY-EDIT-OPENED'
+  );
```

Comment:
I am not sure if we want this in session storage instead of local storage. Any specific reason you picked session storage?

Reply:
> 

Tbh, I decided to use session storage because this data is okay to be deleted after every session. But, regarding your comment I removed it.